### DISCUSSION TITLE
progress_val_score for multiple metrics?

### DISCUSSION THREAD

**Discussion by ZahirBilal** [online-ml/river#1015] [Q&A] [answered]
Hello guys,

I have a question related to my research: Is it possible to evaluate an algorithm using `progressive_val_score `but for multiple metrics?

I tried the following:
![image](https://user-images.githubusercontent.com/87704717/188245267-09d09cc6-687b-4649-9f37-421b81bd8c8f.png)

because I needed to get the scores (gmean, recall) at all the intermediate steps 

![image](https://user-images.githubusercontent.com/87704717/188245286-7e37f6a6-76d5-4de2-b64e-9f2eff095e74.png)

which I got using the code snippet above. but the problem is that calculating the `recall `(final value) using confusion matrix has given different recall value than what I got from `res_r `
where `recall `= 65.9%  and `res_r `(recall) = 81.88% 
is it because the model has been trained and tested twice on the same dataset? I suspected this because when I looked at the `progress_recall.log` it has started measuring the memory consumption starting from [500] Recall: 0.00% – 00:00:00 – 7.49 MB which is almost where the first evaluation process (`progress_gmean.log`) stopped- I am not sure if the model would be discarded after the evaluation ends. 
Please advice :)

**Comment by MaxHalford** [comment, marked as answer]
Hey there Zahir 👋

You can use the `+` operator to use more than metric. For instance:

```py
from river import compose
from river import datasets
from river import evaluate
from river import linear_model
from river import metrics
from river import preprocessing

dataset = datasets.TrumpApproval()

model = compose.Pipeline(
    preprocessing.StandardScaler(),
    linear_model.LinearRegression()
)

metric = metrics.MSE() + metrics.MAE()

evaluate.progressive_val_score(dataset, model, metric)
```

```
MSE: 17.025685, MAE: 1.347434
```

> is it because the model has been trained and tested twice on the same dataset?

Yes indeed, that's the issue. We could have made it so `progressive_val_score` uses a fresh copy of the model, but we didn't on purpose. For instance, if you warmed up your model on some preliminary data, then you wouldn't want to refresh your model.

I hope that helps!

**Reply by ZahirBilal** [reply]
Thanks Max, that has worked perfectly ;)
